src/delta.rs

```diff
@@ -253,30 +257,34 @@ fn handle_hunk_line(painter: &mut Painter, line: &str, state: State, config: &Co
     }
 }
 
-/// Replace initial -/+ character with ' ', expand tabs as spaces, and terminate with newline.
+/// Replace initial -/+ character with ' ', expand tabs as spaces, and optionally terminate with
+/// newline.
 // Terminating with newline character is necessary for many of the sublime syntax definitions to
 // highlight correctly.
 // See https://docs.rs/syntect/3.2.0/syntect/parsing/struct.SyntaxSetBuilder.html#method.add_from_folder
-fn prepare(line: &str, tab_width: usize) -> String {
+fn prepare(line: &str, tab_width: usize, append_newline: bool) -> String {
+    let terminator = if append_newline { "\n" } else { "" };
     if !line.is_empty() {
-        let mut line = line.graphemes(true).peekable();
-        let mut left_fill = "".to_string();
+        let mut line = line.graphemes(true);
+        let mut output_line;
```

Comment:
I get a warning here:

```
warning: variable does not need to be mutable
   --> src\delta.rs:269:13
    |
269 |         let mut output_line;
    |             ----^^^^^^^^^^^
    |             |
    |             help: remove this `mut`
    |
```

If an older version or `rustc` requires the `mut`, you could change the assignment to:

```
let output_line = if tab_width > 0 { ... } else { ... }
```